scripts/generate_plugindata/main.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"html/template"
+	"log" //nolint:revive
+	"os"
+	"time"
+
+	"github.com/gomarkdown/markdown"
+	"github.com/gomarkdown/markdown/ast"
+	"github.com/gomarkdown/markdown/parser"
+)
+
+type PluginData struct {
+	SampleConfig string
+	Description  string
+}
+
+func extractPluginData() (PluginData, error) {
+	readMe, err := os.ReadFile("README.md")
+	if err != nil {
+		return PluginData{}, err
+	}
+	extensions := parser.CommonExtensions | parser.AutoHeadingIDs
+	mdParser := parser.NewWithExtensions(extensions)
+	md := markdown.Parse(readMe, mdParser)
+
+	var p PluginData
+	descriptionHeader := "short-description"
+	configHeader := "configuration"
+	var currentSection string
+
+	for _, t := range md.GetChildren() {
+		switch tok := t.(type) {
+		case *ast.Heading:
+			currentSection = tok.HeadingID
+		case *ast.CodeBlock:
+			if currentSection == configHeader && string(tok.Info) == "toml" {
+				p.SampleConfig = string(tok.Literal)
+			}
+		case *ast.Paragraph:
+			if currentSection != descriptionHeader {
+				continue
+			}
+			for _, c := range tok.Children {
+				if text, ok := c.(*ast.Text); ok {
+					p.Description = string(text.Literal)
+				}
+			}
+		}
+	}
+
+	return p, nil
+}
+
+func generatePluginData(p PluginData) error {
+	generatedTemplate := template.Must(template.New("").Parse(`// Code generated by go generate; DO NOT EDIT; DO NOT COMMIT.
+// This file was generated by robots at
+// {{ .Timestamp }}
+package {{ .GOPACKAGE }}
+
+func (c *CPUStats) LoadGeneratedData() {
```

Comment:
cpu.go
```go
var SampleConfig string

func (c *CPUStats) SampleConfig() string {
	return SampleConfig
}
```

plugindata.go
```go
func LoadGeneratedData() {
	SampleConfig = `# Read metrics about cpu usage
[[inputs.cpu]]
  ## Whether to report per-cpu stats or not
  percpu = true
  ## Whether to report total system cpu stats or not
  totalcpu = true
  ## If true, collect raw CPU time metrics
  collect_cpu_time = false
  ## If true, compute and report the sum of all non-idle CPU states
  report_active = false
`
}
```

Something like this perhaps?

Reply:
Thinking about it, we could even have the actual source file be the template and do something like this:

```
func (c *CPUStats) SampleConfig() string {
	return `{{ .SampleConfig }}`
}
```

Then the go generate will update the main source file, maybe temporarily renaming the original to `cpu.go.tmp`, and then when its done building we can have a clean up step. Not sure if that would just over complicate things.

Reply:
I implemented the idea to see how it might work, and I think it might be a cleaner approach. 
I've pushed the changes for you to look at but to summarize:

Each plugin would have a header like this:
```go
//go:generate go run ../../../scripts/generate_plugindata/main.go
//go:generate go run ../../../scripts/generate_plugindata/main.go --clean
```
Then the sample config function would be updated like this:
```go
func (c *CPUStats) SampleConfig() string {
	return `{{ .SampleConfig }}`
}
```

and if you run `make` you would see this output:
```
$ make
go mod download -x
go generate -run="plugindata/main.go$" ./plugins/inputs/...
go generate -run="plugindata/main.go$" ./plugins/outputs/...
go generate -run="plugindata/main.go$" ./plugins/processors/...
go generate -run="plugindata/main.go$" ./plugins/aggregators/...
go build -ldflags " -X main.commit=7d8fc6f0 -X main.branch=generatesampleconfig -X main.goos=linux -X main.goarch=amd64 -X main.version=1.23.0-7d8fc6f0" ./cmd/telegraf
go generate -run="plugindata/main.go --clean" ./plugins/inputs/...
go generate -run="plugindata/main.go --clean" ./plugins/outputs/...
go generate -run="plugindata/main.go --clean" ./plugins/processors/...
go generate -run="plugindata/main.go --clean" ./plugins/aggregators/...
```

Reply:
I was thinking of setting a variable with an init func so we don't have to call LoadGeneratedData() anywhere

Generated file:
```golang
package CPUStats
var sampleConfig string

func init() {
	sampleConfig = `# Read metrics about cpu usage
[[inputs.cpu]]
  ## Whether to report per-cpu stats or not
  percpu = true
...
`
}
```

Handwritten file:
```golang
package CPUStats
type CPUStats struct{}

func (c *CPUStats) SampleConfig() string {
	return sampleConfig
}
```

But I think I like your idea of directly generating `func SampleConfig() string` more. Doing it that way makes it nearly impossible for the readme to be different than the string SampleConfig() returns.

Reply:
Moving the results of this conversation to the description.